crates/runtime/benches/bench.rs

```diff
@@ -64,103 +85,157 @@ async fn main() -> Result<(), String> {
         upload_results_dataset = Some(env_var);
     }
 
-    let connectors = vec![
-        "spice.ai",
-        "s3",
-        #[cfg(feature = "spark")]
-        "spark",
-        #[cfg(feature = "postgres")]
-        "postgres",
-        #[cfg(feature = "mysql")]
-        "mysql",
-        #[cfg(feature = "odbc")]
-        "odbc-databricks",
-        #[cfg(feature = "odbc")]
-        "odbc-athena",
-        #[cfg(feature = "delta_lake")]
-        "delta_lake",
-    ];
-
-    let mut display_records = vec![];
-
-    for connector in connectors {
-        let (mut benchmark_results, mut rt) =
-            setup::setup_benchmark(&upload_results_dataset, connector, None).await;
+    let args = BenchArgs::parse();
```

Comment:
I think this logic is clearer with using a top-level match across the parameters:

```rust

    match (args.connector.as_deref(), args.accelerator.as_deref(), args.mode.as_deref()) {
        (None, None, None) => {
            // Run all connector / accelerator benchmark tests
            let connectors = vec![
                "spice.ai",
                "s3",
                #[cfg(feature = "spark")]
                "spark",
                #[cfg(feature = "postgres")]
                "postgres",
                #[cfg(feature = "mysql")]
                "mysql",
                #[cfg(feature = "odbc")]
                "odbc-databricks",
                #[cfg(feature = "odbc")]
                "odbc-athena",
                #[cfg(feature = "delta_lake")]
                "delta_lake",
            ];
            for connector in connectors {
                run_connector_bench(connector, &upload_results_dataset).await?;
            }
            let accelerators: Vec<Acceleration> = vec![
                create_acceleration("arrow", acceleration::Mode::Memory),
                #[cfg(feature = "duckdb")]
                create_acceleration("duckdb", acceleration::Mode::Memory),
                #[cfg(feature = "duckdb")]
                create_acceleration("duckdb", acceleration::Mode::File),
                #[cfg(feature = "sqlite")]
                create_acceleration("sqlite", acceleration::Mode::Memory),
                #[cfg(feature = "sqlite")]
                create_acceleration("sqlite", acceleration::Mode::File),
            ];
            for accelerator in accelerators {
                run_accelerator_bench(accelerator, &upload_results_dataset).await?;
            }
        },
        (Some(connector), None, None) => {
            // Run connector benchmark test
            run_connector_bench(connector, &upload_results_dataset).await?;
        },
        (None, Some(accelerator), mode) => {
            // Run accelerator benchmark test
            let mode = match mode {
                Some("file") => Mode::File,
                Some("memory") | None => Mode::Memory,
                _ => return Err(format!("Invalid mode parameter for {accelerator} accelerator")),
            };
            let acceleration = create_acceleration(accelerator, mode);
            run_accelerator_bench(acceleration, &upload_results_dataset).await?;
        },
        _ => return Err("Invalid command line input: accelerator or mode parameter supplied for connector benchmark".to_string()),
    }
```